File: leetcode/剑指/剑指.10.青蛙跳台阶问题.go
var Fib map[int]int = map[int]int{0: 1, 1: 1}
func numWays(n int) int {
    if _, ok := Fib[n];ok {
        return Fib[n];
    }
    Fib[n] = (numWays(n - 1) + numWays(n - 2)) % 1000000007;
    return Fib[n];
}

File: leetcode/剑指/剑指.10.青蛙跳台阶问题.py
class Solution:
    Ways = {0: 1, 1: 1}
    def numWays(self, n: int) -> int:
        if n in self.Ways:
            return self.Ways[n]
        self.Ways[n] = (self.numWays(n - 1) + self.Ways[n - 2]) % 1000000007
        return self.Ways[n]